File: 605.can-place-flowers.py
#
# @lc app=leetcode id=605 lang=python3
#
# [605] Can Place Flowers
#

# @lc code=start
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
        idx = 0
        available = 0
        flowerbed_size = len(flowerbed)

        while (idx < flowerbed_size):
            prev = 0 if idx == 0 else flowerbed[idx - 1]
            next = 0 if idx == flowerbed_size - 1 else flowerbed[idx + 1]
            if flowerbed[idx] == 0 and prev == 0 and next == 0:
                available += 1
                idx += 2
            else:
                idx += 1
            
            if available >= n:
                return True
        
        return False

                
# @lc code=end

File: 605.can-place-flowers.java
/*
 * @lc app=leetcode id=605 lang=java
 *
 * [605] Can Place Flowers
 */

// @lc code=start
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        
        int currentIdx = 0;
        int availables = 0;
        
        int prev = 0;
        int next = 0;

        while (currentIdx < flowerbed.length) {
            prev = currentIdx == 0 ? 0 : flowerbed[currentIdx - 1];
            next = currentIdx == flowerbed.length - 1 ? 0 : flowerbed[currentIdx + 1];

            if (flowerbed[currentIdx] == 0 && prev == 0 && next == 0) {
                availables++;
                currentIdx += 2;

            } else {
                currentIdx += 1;
            }

            if (availables >= n) {
                    return true;
            }
        }

        return false;
    }
}
// @lc code=end

